features/step_definitions/back_office/edit_steps.rb

```diff
@@ -0,0 +1,99 @@
+require "pry"
+
+When("I edit the most recent registration") do
+  @edited_info = {
+    contact_email: generate_email
+  }
+
+  @bo.dashboard_page.view_reg_details(search_term: @reg_number)
+
+  registration_details_page = RegistrationDetailsPage.new
```

Comment:
RegistrationDetailsPage.new should have already been defined here as part of the original sign in step (`load_all_apps`).
This only needs to be a single line:
`@bo.registration_details_page.edit_link.click`

Similar comment for other Page.new definitions.